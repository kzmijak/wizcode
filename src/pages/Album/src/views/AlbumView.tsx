import { albumState, useAlbumsFetch } from "modules/AlbumsState";
import { FC } from "react";
import { Navigate, useParams } from "react-router";
import { useRecoilValue } from "recoil";
import { PATH_APP } from "routes/src/paths";
import { AlbumInfo } from "../components/AlbumInfo";
import { Page } from "components/Page";

export const AlbumView: FC = () => {
  useAlbumsFetch();

  const { albumId } = useParams();
  const album = useRecoilValue(albumState(albumId));

  if (!album) return <Navigate to={PATH_APP.notFound} replace />;
  return (
    <Page title={album.title}>
      <AlbumInfo {...album} />
    </Page>
  );
};
