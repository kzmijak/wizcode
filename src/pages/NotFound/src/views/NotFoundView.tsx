import { Stack, Typography } from "@mui/joy";
import { Page } from "components/Page";
import { FC } from "react";

export const NotFoundView: FC = () => {
  return (
    <Page title="404">
      <Stack flexGrow={1} justifyContent="center" alignItems="center">
        <Typography level="display1">Whoopsie Daisy!</Typography>
        <Typography level="h4">
          Resource not found, sorry about that{" "}
        </Typography>
      </Stack>
    </Page>
  );
};
