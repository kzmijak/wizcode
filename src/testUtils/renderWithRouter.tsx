import { render } from "@testing-library/react";
import { RouterProvider, createMemoryRouter } from "react-router";

export const renderWithRouter = (ui: JSX.Element, ...location: string[]) => {
  const router = createMemoryRouter([{ path: "*", element: ui }], {
    initialEntries: ["", ...location],
  });

  return render(<RouterProvider router={router} />);
};
