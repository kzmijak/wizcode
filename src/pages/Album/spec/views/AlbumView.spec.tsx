import { describe, it, expect } from "vitest";
import { AlbumView } from "../../src/views/AlbumView";
import { renderWithRouter } from "testUtils/renderWithRouter";

describe("AlbumView", () => {
  const Component = () => <AlbumView />;

  it("should match snapshot", () => {
    expect(renderWithRouter(<Component />)).toMatchSnapshot();
  });
});
