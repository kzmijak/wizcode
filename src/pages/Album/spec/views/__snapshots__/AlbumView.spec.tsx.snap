// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AlbumView > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2>
        Unexpected Application Error!
      </h2>
      <h3
        style="font-style: italic;"
      >
        This component must be used inside a &lt;RecoilRoot&gt; component.
      </h3>
      <pre
        style="padding: 0.5rem; background-color: rgba(200, 200, 200, 0.5);"
      >
        Error: This component must be used inside a &lt;RecoilRoot&gt; component.
    at err (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:22:17)
    at Object.notInAContext (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:4182:9)
    at updateRetainCount (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:3336:21)
    at useRetain_ACTUAL (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:4761:7)
    at useRetain (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:4719:10)
    at Proxy.useRecoilValueLoadable (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:5347:5)
    at Module.useAlbumsFetch (C:\\Users\\kzmij\\Desktop\\wizcode\\src\\modules\\AlbumsState\\src\\hooks\\useAlbumsFetch.ts:8:26)
    at AlbumView (C:\\Users\\kzmij\\Desktop\\wizcode\\src\\pages\\Album\\src\\views\\AlbumView.tsx:10:3)
    at renderWithHooks (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\react-dom\\cjs\\react-dom.development.js:16305:18)
    at mountIndeterminateComponent (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\react-dom\\cjs\\react-dom.development.js:20074:13)
      </pre>
      <p>
        ðŸ’¿ Hey developer ðŸ‘‹
      </p>
      <p>
        You can provide a way better UX than this when your app throws errors by providing your own 
        <code
          style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
        >
          ErrorBoundary
        </code>
         or
         
        <code
          style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
        >
          errorElement
        </code>
         prop on your route.
      </p>
    </div>
  </body>,
  "container": <div>
    <h2>
      Unexpected Application Error!
    </h2>
    <h3
      style="font-style: italic;"
    >
      This component must be used inside a &lt;RecoilRoot&gt; component.
    </h3>
    <pre
      style="padding: 0.5rem; background-color: rgba(200, 200, 200, 0.5);"
    >
      Error: This component must be used inside a &lt;RecoilRoot&gt; component.
    at err (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:22:17)
    at Object.notInAContext (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:4182:9)
    at updateRetainCount (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:3336:21)
    at useRetain_ACTUAL (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:4761:7)
    at useRetain (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:4719:10)
    at Proxy.useRecoilValueLoadable (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\recoil\\cjs\\index.js:5347:5)
    at Module.useAlbumsFetch (C:\\Users\\kzmij\\Desktop\\wizcode\\src\\modules\\AlbumsState\\src\\hooks\\useAlbumsFetch.ts:8:26)
    at AlbumView (C:\\Users\\kzmij\\Desktop\\wizcode\\src\\pages\\Album\\src\\views\\AlbumView.tsx:10:3)
    at renderWithHooks (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\react-dom\\cjs\\react-dom.development.js:16305:18)
    at mountIndeterminateComponent (C:\\Users\\kzmij\\Desktop\\wizcode\\node_modules\\react-dom\\cjs\\react-dom.development.js:20074:13)
    </pre>
    <p>
      ðŸ’¿ Hey developer ðŸ‘‹
    </p>
    <p>
      You can provide a way better UX than this when your app throws errors by providing your own 
      <code
        style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
      >
        ErrorBoundary
      </code>
       or
       
      <code
        style="padding: 2px 4px; background-color: rgba(200, 200, 200, 0.5);"
      >
        errorElement
      </code>
       prop on your route.
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
